src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -1579,6 +1579,115 @@ public void TestConnectionStringEnvVarHasHighestPrecedence()
             }
         }
 
+        /// <summary>
+        /// Sanity check to validate that DAB engine works fine when used with a config file without the nested 
+        /// mutations feature flag section.
+        /// The runtime graphql section of the config file used looks like this: 
+        ///
+        /// "graphql": {
+        ///    "path": "/graphql",
+        ///    "allow-introspection": true
+        ///  }
+        /// 
+        /// Without the nested mutations feature flag section, DAB engine should be able to 
+        ///  1. Successfully deserialize the config file.
+        ///  2. Process REST and GraphQL API requests.
+        /// 
+        /// </summary>
+        [TestMethod]
+        [TestCategory(TestCategory.MSSQL)]
+        public async Task SanityTestForRestAndGQLRequestsWithoutNestedMutationFeatureFlagSection()
+        {
+            // Hard-coded json string for Book entity
+            string entityJson = @"
+            {
+              ""entities"": {
+                    ""Book"": {
+                    ""source"": {
+                        ""object"": ""books"",
+                        ""type"": ""table""
+                    },
+                    ""graphql"": {
+                        ""enabled"": true,
+                        ""type"": {
+                        ""singular"": ""book"",
+                        ""plural"": ""books""
+                        }
+                    },
+                    ""rest"":{
+                        ""enabled"": true
+                    },
+                    ""permissions"": [
+                        {
+                        ""role"": ""anonymous"",
+                        ""actions"": [
+                                {
+                                    ""action"": ""read""
+                                }
+                            ]
+                        }
+                    ],
+                    ""mappings"": null,
+                    ""relationships"": null
+                    }
+                }
+            }";
+
+            // The config file is constructed by merging the hard-coded json strings to mimic the scenario where config file is
+            // hand-edited (instead of using CLI) by the users.
+            string configJson = TestHelper.AddPropertiesToJson(TestHelper.BASE_CONFIG, entityJson);
+            RuntimeConfigLoader.TryParseConfig(configJson, out RuntimeConfig deserializedConfig, logger: null, GetConnectionStringFromEnvironmentConfig(environment: TestCategory.MSSQL));
```

Comment:
tests which validate the results of `RuntimeConfigLoader.TryParseConfig(...)` would be helpful to know how the new nested insert config behaves when unexpected values are assigned or not assigned:
- nulls assigned to properties, 
```json
{
  "nested-mutations": null
}
```
- empty objects 
```json
{
  "nested-mutations": {
    "create": {}
  }
}

{
  "nested-mutations": {}
}
```

Reply:
Ack. Looking into this.

Reply:
I've added tests to validate these scenarios: `ConfigurationTests.ValidateDeserializationOfConfigWithNullOrEmptyInvalidNestedMutationSection()`